thetis/solver2d.py

```diff
@@ -592,6 +599,16 @@ def create_timestepper(self):
                 steppers[self.options.nh_model_options.free_surface_timestepper_type]
             )
         elif self.solve_tracer:
+            if self.options.tracer_priorities != {}:
+                reordered = OrderedDict()
+                priorities = list(self.options.tracer_priorities.keys())
+                priorities.sort()
+                for priority in priorities:
+                    label = self.options.tracer_priorities[priority]
+                    reordered[label] = self.options.tracer.pop(label)
+                for label in list(self.options.tracer.keys()):
+                    reordered[label] = self.options.tracer.pop(label)
+                self.options.tracer = reordered
```

Comment:
Ah no (there's no separate ordering) - overwriting user options is never a good idea though: `add_tracer_2d` is a nice helper function that makes it easier to add tracer fields, but we should still treat the options dictionaries (FrozenHasTraits) as user provided input data that we're not allowed to clobber.

Reply:
Yeah good point. I was mainly thinking of the (not yet supported) case where you might want to solve a subset of the tracer equations as a mixed system. I'll drop this `priority` business, for now.